BeVolt/Drivers/src/LTC6811.c

```diff
@@ -75,7 +75,185 @@ Copyright 2017 Linear Technology Corp. (LTC)
+  cmd_index = 4;
+	
+  for (uint8_t current_ic = total_ic; current_ic > 0; current_ic--){      // executes for each LTC6811 in daisy chain, this loops starts with
+    // the last IC on the stack. The first configuration written is
+    // received by the last IC in the daisy chain
+
+    for (uint8_t current_byte = 0; current_byte < BYTES_IN_REG; current_byte++){
+      cmd[cmd_index] = data[((current_ic-1)*6)+current_byte];
+      cmd_index = cmd_index + 1;
+    }
+
+    data_pec = (uint16_t)pec15_calc(BYTES_IN_REG, &data[(current_ic-1)*6]);    // calculating the PEC for each Iss configuration register data
+    cmd[cmd_index] = (uint8_t)(data_pec >> 8);
+    cmd[cmd_index + 1] = (uint8_t)data_pec;
+    cmd_index = cmd_index + 2;
+  }
+	
+  cs_set(0);
+  SPI1_WriteMulti(cmd, CMD_LEN);
+  cs_set(1);
+}
+
+/* read_68
+ * Generic function to write 6811 commands and read data. Function calculated PEC for tx_cmd data
+ * @param unsigned int 8-bit number of ICs in the daisy chain
+ * @param 2 Byte array containing the BMS command to be sent
+ * @param Array that the read back data will be stored
+ * @return 8-bit int PEC Status
+ */
+int8_t read_68(uint8_t total_ic, uint8_t tx_cmd[2], uint8_t *rx_data){
+  const uint8_t BYTES_IN_REG = 8;
+  uint8_t cmd[4];
+  uint8_t data[256];
+  int8_t pec_error = 0;
+  uint16_t cmd_pec;
+  uint16_t data_pec;
+  uint16_t received_pec;
+
+  // data = (uint8_t *) malloc((8*total_ic)*sizeof(uint8_t)); // This is a problem because it can fail
+
+  cmd[0] = tx_cmd[0];
+  cmd[1] = tx_cmd[1];
+  cmd_pec = pec15_calc(2, cmd);
+  cmd[2] = (uint8_t)(cmd_pec >> 8);
+  cmd[3] = (uint8_t)(cmd_pec & 0x00FF);
+
+  cs_set(0);
+  SPI1_WriteReadMulti(cmd, 4, data, (BYTES_IN_REG*total_ic), true);         //Read the configuration data of all ICs on the daisy chain into
+  cs_set(1);                          //rx_data[] array
+
+  for (uint8_t current_ic = 0; current_ic < total_ic; current_ic++){       //executes for each LTC6811 in the daisy chain and packs the data
+    //into the r_comm array as well as check the received Config data
+    //for any bit errors
+    for (uint8_t current_byte = 0; current_byte < BYTES_IN_REG; current_byte++){
+      rx_data[(current_ic*8)+current_byte] = data[current_byte + (current_ic*BYTES_IN_REG)];
+    }
+    received_pec = (rx_data[(current_ic*8)+6]<<8) + rx_data[(current_ic*8)+7];
+    data_pec = pec15_calc(6, &rx_data[current_ic*8]);
+    if (received_pec != data_pec){
+      pec_error = -1;
+    }
+  }
+
+  return(pec_error);
+}
+
+
+/* pec15_calc
+ * Calculates  and returns the CRC15
+ * @param int 32-bit number of bytes that will be used to calculate a PEC
+ * @param array of data that will be used to calculate a PEC
+ * @returns the calculated pec15 as an unsigned int
+ */
+uint16_t pec15_calc(int32_t len, uint8_t *data){
+  uint16_t remainder,addr;
+
+  remainder = 16;//initialize the PEC
+  for (uint8_t i = 0; i<len; i++){ // loops for each byte in data array
```

Comment:
Everything here is just ugly.

Reply:
I mean that some spaces, capitalization, and parenthesis could go a long way for some of these statements.